src/haliax/specialized_fns.py

```diff
@@ -0,0 +1,19 @@
+import jax
+import jax.numpy as jnp
+
+from .axis import Axis, AxisSelector
+from .core import NamedArray
+
+
+def top_k(arr: NamedArray, axis: AxisSelector, k: int) -> NamedArray:
+    pos = arr._lookup_indices(axis)
+    if pos is None:
+        raise ValueError(f"Axis {axis} not found in {arr}")
+    new_array = jnp.moveaxis(arr.array, pos, -1)  # move axis to the last position
+    values, _ = jax.lax.top_k(new_array, k=k)
+    values = jnp.moveaxis(values, -1, pos)  # move axis back to its original position
+
+    axis = arr.resolve_axis(axis)
+    new_axis = Axis(f"top_{k}_of_{axis.name}", size=k)
```

Comment:
imho i think it's better to just keep the same name and resize it, or even better:

```
def top_k(arr: NamedArray, axis: AxisSelector, k: int, new_axis: Optional[AxisSelector] = None) -> NamedArray:
  ...
  if new_axis is None: 
       new_axis = axis.resize(k)
  else:
      ... 
 
```

Otherwise it's a pain to refer to the top-k axis

Reply:
Yup, this makes sense – should be fixed now!